src/libostree/ostree-repo-pull.c

```diff
@@ -1534,11 +1498,13 @@ process_one_static_delta_fallback (OtPullData   *pull_data,
         {
           if (!g_hash_table_lookup (pull_data->requested_metadata, checksum))
             {
-              gboolean do_fetch_detached;
               g_hash_table_insert (pull_data->requested_metadata, checksum, checksum);
               
-              do_fetch_detached = (objtype == OSTREE_OBJECT_TYPE_COMMIT);
-              enqueue_one_object_request (pull_data, checksum, objtype, do_fetch_detached, FALSE);
+              /* Fetch metadata */
+              if (objtype == OSTREE_OBJECT_TYPE_COMMIT)
+                fetch_object (pull_data, g_strdup (checksum), OSTREE_OBJECT_TYPE_COMMIT_META);
```

Comment:
It's only a leak if fetch_object is not transfer-full. I have added a patch to make that so.